### PR TITLE
Adding user guide for embedded Service Binding

### PR DESCRIPTION
**Pull request by navidsh**
**What this PR does / why we need it?**:
- Adding documentation for embedded Service Binding user definition

**Does this PR introduce a user-facing change?**

- [x] User guide
- [x] `CHANGELOG.md`

**Which issue(s) this PR fixes**: #100

### CODE REVIEW

File: doc/user-guide.adoc [hunk 1]
```diff
@@ -414,9 +453,9 @@ spec:
     resourceRef: my-binding
 ----
 
-After the Service Binding Operator processes the `my-binding` custom resource of the `ServiceBindingRequest` type, it creates a `Secret` object. The `Secret` resource contains binding information that the backing service provides. The Runtime Component Operator injects the `Secret` resource as environment variables into application pods.
+Similar to what was described above, after the Service Binding Operator processes the `my-binding` custom resource of the `ServiceBindingRequest` type, it creates a `Secret` object. The `Secret` resource contains binding information that the backing service provides. The Runtime Component Operator injects the `Secret` resource as environment variables into application pods.
```
Comment (chirp1) [inline doc/user-guide.adoc:456]: - Change "Similar to what was described above, after the Service Binding Operator processes..." to "After the Service Binding Operator, as defined previously, processes...".
- Change "the `Secret` resource as environment variables into " to "the `Secret` resource as an environment variable into ".

File: doc/user-guide.adoc [hunk 2]
```diff
@@ -62,8 +62,9 @@ Each `RuntimeComponent` CR must at least specify the `applicationImage` paramete
 | `initContainers` | The list of link:++https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.14/#container-v1-core++[Init Container] definitions.
 | `sidecarContainers` | The list of `sidecar` containers. These are additional containers to be added to the pods. Note: Sidecar containers should not be named `app`.
 | `architecture` | An array of architectures to be considered for deployment. Their position in the array indicates preference.
-| `bindings.autoDetect` | A boolean to toggle whether the operator should automatically detect and use a `ServiceBindingRequest` resource with the same name as the CR. The default value for this parameter is `true`.
-| `bindings.resourceRef` | The name of a `ServiceBindingRequest` resource created manually in the same namespace as the application.
+| `bindings.embedded` | A YAML object that represents a `ServiceBindingRequest` custom resource.
+| `bindings.autoDetect` | A boolean to toggle whether the operator should automatically detect and use a `ServiceBindingRequest` custom resource with the same name as the CR. The default value for this parameter is `true`.
```
Comment (leochr) [inline doc/user-guide.adoc:66]: _custom resource with the same name as the CR_
Needs update due to the suffix change

File: doc/user-guide.adoc [hunk 3]
```diff
@@ -400,7 +401,45 @@ spec:
...
+            kind: Kafka
+            resourceRef: my-cluster
+            version: v1beta1
+----
+
+After the Runtime Component Operator processes the `RuntimeComponent` custom resource, it creates a `ServiceBindingRequest` custom resource named `my-app`.
+
+After the Service Binding Operator processes the `my-app` custom resource of the `ServiceBindingRequest` type, it creates a `Secret` object. The `Secret` resource contains binding information that the backing service provides. The Runtime Component Operator injects the `Secret` resource as an environment variable into application pods.
+
+The YAML definition specified in the `bindings.embedded` parameter must not include a `metadata`section because the Runtime Component operator adds a `metadata`section. If the YAML definition does not include an `apiVersion` field or a `kind` field, the Runtime Component operator uses the default `GroupVersionKind` value specified in the Operator Config Map.
```
Comment (leochr) [inline doc/user-guide.adoc:438]: A space is needed: `metadata`section --> `metadata` section

File: doc/user-guide.adoc [hunk 4]
```diff
@@ -400,7 +401,45 @@ spec:
...
+      kind: ServiceBindingRequest
+      spec:
+        backingServiceSelectors:
+          - group: kafka.strimzi.io
+            kind: Kafka
+            resourceRef: my-cluster
+            version: v1beta1
+----
+
+After the Runtime Component Operator processes the `RuntimeComponent` custom resource, it creates a `ServiceBindingRequest` custom resource named `my-app`.
```
Comment (leochr) [inline doc/user-guide.adoc:434]: Some updates needed due to the suffix.

File: doc/user-guide.adoc [hunk 5]
```diff
@@ -400,7 +401,45 @@ spec:
 
 The link:++https://github.com/redhat-developer/service-binding-operator++[Service Binding Operator] enables application developers to bind applications together with operator-managed backing services. This can be achieved by creating a `ServiceBindingRequest` custom resource.
 
-For the Runtime Component Operator to pass the binding information to the application, you can specify the name of the `ServiceBindingRequest` custom resource as follows:
+
+For the Runtime Component Operator to pass the binding information to the application, define your `ServiceBindingRequest` custom resource in one of the following two ways:
+
+. Define the `ServiceBindingRequest` custom resource YAML within the `bindings.embedded` parameter in your `RuntimeComponent` custom resource.
+. Create the `ServiceBindingRequest` custom resource manually and either refer to it explicitly in your `RuntimeComponent` custom resource or let the operator detect it automatically.
```
Comment (leochr) [inline doc/user-guide.adoc:408]: It may be good to clarify that it'll be auto detected only if the naming convention is followed.

**Comment by leochr** [comment]
@navidsh Looks good for the most part. Some updates needed due to the suffix changes. Thanks.

**Pull request merged**